modules/config/home-cursor.nix

```diff
@@ -0,0 +1,158 @@
+
+  imports = [
+    (mkAliasOptionModule [ "xsession" "pointerCursor" "package" ] [
+      "home"
+      "pointerCursor"
+      "package"
+    ])
+    (mkAliasOptionModule [ "xsession" "pointerCursor" "name" ] [
+      "home"
+      "pointerCursor"
+      "name"
+    ])
+    (mkAliasOptionModule [ "xsession" "pointerCursor" "size" ] [
+      "home"
+      "pointerCursor"
+      "size"
+    ])
+    (mkAliasOptionModule [ "xsession" "pointerCursor" "defaultCursor" ] [
+      "home"
+      "pointerCursor"
+      "x11"
+      "defaultCursor"
+    ])
+
+    ({ ... }: {
+      warnings = optional (any (x:
+        getAttrFromPath
+        ([ "xsession" "pointerCursor" ] ++ [ x ] ++ [ "isDefined" ])
+        options) [ "package" "name" "size" "defaultCursor" ]) ''
+          The option `xsession.pointerCursor` has been merged into `home.pointerCursor` and will be removed
+          in the future. Please change to set `home.pointerCursor` directly and enable `home.pointerCursor.x11.enable`
+          to generate x11 specific cursor configurations. You can refer to the documentation for more details.
+        '';
+    })
+  ];
+
+  options = {
+    home.pointerCursor = mkOption {
+      type = types.nullOr pointerCursorModule;
+      default = null;
+      description = ''
+        Cursor configuration. Set to <literal>null</literal> to disable.
+        </para><para>
+        Top-level options declared under this submodule are backend indepedent
+        options. Options declared under namespaces such as <literal>x11</literal>
+        are backend specific options. By default, only backend independent cursor
+        configurations are generated. If you need configurations for specific
+        backends, you can toggle them via the enable option. For example, <varname>
+        <link linkend="opt-home.pointerCursor.x11.enable">home.pointerCursor.x11.enable</link>
+        </varname> will enable x11 cursor configurations.
+      '';
+    };
+  };
+
+  config = mkIf (cfg != null) (mkMerge [
+    {
+      assertions = [
+        (hm.assertions.assertPlatform "home.pointerCursor" pkgs platforms.linux)
+      ];
+
+      home.packages = [ cfg.package ];
+
+      # Set name in icons theme, for compatibility with AwesomeWM etc. See:
+      # https://github.com/nix-community/home-manager/issues/2081
+      # https://wiki.archlinux.org/title/Cursor_themes#XDG_specification
+      home.file.".icons/default/index.theme".text = ''
+        [icon theme]
+        Name=Default
+        Comment=Default Cursor Theme
+        Inherits=${cfg.name}
+      '';
+
+      # Set directory to look for cursors in, needed for some applications
+      # that are unable to find cursors otherwise. See:
+      # https://github.com/nix-community/home-manager/issues/2812
+      # https://wiki.archlinux.org/title/Cursor_themes#Environment_variable
+      home.sessionVariables = {
+        XCURSOR_PATH = mkDefault (concatStringsSep ":" [
+          "\${XCURSOR_PATH}"
```

Comment:
See my comment in the other PR https://github.com/nix-community/home-manager/pull/2902#discussion_r853390670